code/modules/ghostroles/instantiator.dm

```diff
@@ -86,6 +86,7 @@
 	return			// tgcode does this automatically
 
 /datum/ghostrole_instantiator/human/random/species
+	var/can_change_appearance = TRUE
```

Comment:
this variable and SetAppearance needs to be on /datum/ghostrole_instantiator/human/random
SetAppearance should probably be renamed to PickAppearance, invoked asynchronously with `INVOKE_ASYNC(src, /datum/ghostrole_instantiator/human/random/proc/PickAppearance, H, params)`. This way, it doesn't interfere with the spawning process. in the pickappearance proc you can then ask the player what they want to be.